dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/NativeRelationAnalysis.java

```diff
@@ -716,10 +761,17 @@ public MutableKnowledge visitReadModifyWrites(ReadModifyWrites rmw) {
                         if (!(load instanceof Load) || intermediaries.stream().anyMatch(e -> exec.isImplied(load, e))) {
                             continue;
                         }
-                        may.add(load, store);
-                        if (intermediaries.stream().allMatch(e -> exec.areMutuallyExclusive(load, e)) &&
-                                (store.doesRequireMatchingAddresses() || alias.mustAlias((Load) load, store))) {
-                            must.add(load, store);
+                        boolean isMust = intermediaries.stream().allMatch(e -> exec.areMutuallyExclusive(load, e)) &&
+                                (store.doesRequireMatchingAddresses() || alias.mustAlias((Load) load, store));
+                        // Idea: For RMWs torn into bytewise transactions,
+                        // matching only occurs between the last load and the first store.
+                        // This implementation builds the complete bipartite graph between both transactions.
+                        Set<Event> st = transactionMap.getOrDefault(store, Set.of(store));
+                        for (Event ld : transactionMap.getOrDefault(load, Set.of(load))) {
```

Comment:
Isn't this wrong? If load and store are split into 4 parts each, you should have exactly 4 LXSX-edges, no?
Your comment (and your code) suggests that you put 4x4=16 edges.